### PR TITLE
VACMS-3465: Add spotlight panel.

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
See #3465 

## Testing done
Visual / Behat

## Screenshots
image
image

## QA steps
- As an administrator, go to `/node/add/campaign_landing_page` and click on the Spotlight panel
- Visually verify that only a `Spotlight opt-in` toggle appears
- Visually verify that clicking it reveals all section fields, and text fields are required
- Add list of link teasers, and visually verify that after third is added, button to add more no longer appears

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [x] `Product Support Team`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by kevwalsh** [comment]
Nice, i have a few suggestions for some changes, minor field labels and help text. 

Create_Campaign_Landing_Page___VA_gov_CMS

Create_Campaign_Landing_Page___VA_gov_CMS

Also, i think we should replace `field_spotlight_cta` with `field_clp_spotlight_cta` (sorry), to bring it in line with the other fields that won't be reused outside this product.

**Review by kevwalsh — CHANGES_REQUESTED** [review]
I reviewed the content model and editorial experience. Not the boolean behavior code. Requested changes above.

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 1]
```diff
@@ -180,6 +180,77 @@ function _va_gov_backend_add_vamc_regions_select(array &$form) {
...
+  // Add clp panels data to this array to enable state management.
+  // Make sure required fields are not set to required in config when using
+  // this pattern.
+  return [
+    'trigger_groups' => [
+      [
+        'trigger' => 'field_clp_spotlight_opt_in',
+        'toggle_fields' => [
+          'field_clp_spotlight_header' => [
+            'required' => TRUE,
```
Comment (oksana-c) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:199]: ❤️ this new structure. thanks for giving it a shot!

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 2]
```diff
@@ -180,6 +180,77 @@ function _va_gov_backend_add_vamc_regions_select(array &$form) {
...
+/**
+ * Used to return clp panel fields and paragraphs.
+ *
+ * @return array
+ *   The clp panel field data.
+ */
+function _va_gov_backend_get_clp_panels() {
+  // Add clp panels data to this array to enable state management.
+  // Make sure required fields are not set to required in config when using
+  // this pattern.
```
Comment (oksana-c) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:192]: 👍

**Review by oksana-c — APPROVED** [review]
Code looks good ⭐ . Didn't retest functionality since devshop is still building. @kevwalsh , leaving it up to you to re-test UI after updates.

**Review by kevwalsh — APPROVED** [review]
QA'd the form and looks good!

File: docroot/modules/custom/va_gov_backend/js/script.js [hunk 3]
```diff
@@ -22,6 +22,15 @@
     }
   };
 
+  Drupal.behaviors.vaGovClpLimitListOfLinks = {
+    attach: function () {
+      // Don't allow more than 3 link teasers in clp spotlight panel.
+        if ($('#field-clp-spotlight-link-teasers-add-more-wrapper .paragraphs-dropbutton-wrapper').length > 3) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/js/script.js:28]: Doesn't this only remove the button if there are 4?  Shouldn't it be >=?

**Review by swirtSJW — COMMENTED** [review]
Waiting to test this until I get devshop sorted out 
image

File: docroot/modules/custom/va_gov_backend/js/script.js (hunk 3 above)
Comment (ethanteague) [reply docroot/modules/custom/va_gov_backend/js/script.js:28]: Good q, but nope. For some reason there is an extra instance of this element being counted via behaviors. I looked up and down in the dom, and couldn't figure out why, but the count was consistently off by one in both brave and chromium, @swirtSJW

File: config/sync/core.entity_form_display.node.campaign_landing_page.default.yml [hunk 4]
```diff
@@ -6,6 +6,11 @@ dependencies:
     - entity_browser.browser.promo_blocks_browser
     - field.field.node.campaign_landing_page.field_administration
     - field.field.node.campaign_landing_page.field_benefit_categories
+    - field.field.node.campaign_landing_page.field_clp_spotlight_cta
+    - field.field.node.campaign_landing_page.field_clp_spotlight_header
+    - field.field.node.campaign_landing_page.field_clp_spotlight_intro_text
+    - field.field.node.campaign_landing_page.field_clp_spotlight_link_teasers
+    - field.field.node.campaign_landing_page.field_clp_spotlight_opt_in
```
[the remark this answers is not in the document]
Comment (ethanteague) [reply config/sync/core.entity_form_display.node.campaign_landing_page.default.yml:13]: Straight out of @kevwalsh 's mind / old pr.

**Comment by swirtSJW** [comment]
This tests out on the other PR tests 
image

**Comment by swirtSJW** [comment]
Proofing issues raised  here https://dsva.slack.com/archives/CT4GZBM8F/p1606772498165900 are already in another ticket.

**Pull request merged**